core/src/apps/common/passphrase.py

```diff
@@ -10,16 +10,34 @@
 if TYPE_CHECKING:
     from trezor.messages import ThpCreateNewSession
 
+if __debug__:
+    from trezor import log
+
 
 def is_enabled() -> bool:
     return storage_device.is_passphrase_enabled()
 
 
+def is_always_on_device() -> bool:
+    return storage_device.get_passphrase_always_on_device()
+
+
 async def get_passphrase(msg: ThpCreateNewSession) -> str:
-    if not is_enabled():
-        return ""
+    # Device setting "disabled passphrase protection" is ignored
+    if __debug__:
+        if not is_enabled() and msg.passphrase:
+            log.warning(
+                __name__,
+                "Creating new session with passphrase, ignoring device settings.",
+            )
+
+    # When always_on_device is True, messages with passphrase raise a DataError
+    if is_always_on_device() and msg.passphrase is not None:
```

Comment:
nit: call `get_passphrase_always_on_device()` once and reuse the result:
```suggestion
    is_always_on_device = storage_device.get_passphrase_always_on_device()
    if is_always_on_device and msg.passphrase is not None:
```